### PR TITLE
Add experimental OkHttp3 websocket client support

### PR DESCRIPTION
**Pull request by rogerhu**
We should deprecate TubeSocket if this works.

@mmimeault

### CODE REVIEW

File: ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+        @Override
+        public void onFailure(okhttp3.WebSocket webSocket, Throwable t, Response response) {
+            webSocketClientCallback.onError(t);
+        }
+    };
+
+    private OkHttp3WebSocketClient(WebSocketClientCallback webSocketClientCallback, URI hostUrl) {
+        this.webSocketClientCallback = webSocketClientCallback;
+        client = new OkHttpClient();
```
Comment (mmimeault) [inline ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java]: since we don't reuse the client, maybe we should only create it in the open() method when creating the websocket?

Comment (rogerhu) [inline ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java]: I've created two constructors now, one default that instantiates one, and another that allows you to pass one in...

Comment (mmimeault) [inline ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java]: Mmm not sure to understand. Maybe my comment wasn't clear.
Why did you add the ability to pass a OkHttpClient? Is it useful?

Comment (rogerhu) [inline ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java]: Yes adding a custom OkHttpClient Is useful for troubleshooting and custom options:

https://code.facebook.com/posts/393927910787513/stetho-a-new-debugging-platform-for-android/

Comment (mmimeault) [inline ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java]: OK thanks for the clarification.

File: ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import okio.ByteString;
+
+/* package */ class OkHttp3WebSocketClient implements WebSocketClient {
+
+    private static final String LOG_TAG = "OkHttpWebSocketClient";
+
+    private final WebSocketClientCallback webSocketClientCallback;
+    private WebSocket webSocket;
+    private State state = State.NONE;
+    private OkHttpClient client;
```
Comment (mmimeault) [inline ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java]: You could make it final.

Comment (rogerhu) [inline ParseLiveQuery/src/main/java/com/parse/OkHttp3WebSocketClient.java]: Fixed
